service/history/queues/queue_base.go

```diff
@@ -338,16 +340,14 @@ func (p *queueBase) checkpoint() {
 	for readerID, reader := range p.readerGroup.Readers() {
 		scopes := reader.Scopes()
 
-		if len(scopes) == 0 {
-			if readerID != DefaultReaderId {
-				p.readerGroup.RemoveReader(readerID)
-			}
+		if len(scopes) == 0 && readerID != DefaultReaderId {
+			p.readerGroup.RemoveReader(readerID)
 			continue
 		}
```

Comment:
Default reader won't be removed, so we still need to update it's progress by adding it to `readerScopes` on L348. 

But we don't need to persist data for a reader if it's empty, so in `updateQueueState` is updated to account for that.